primitives/src/crh/bowe_hopwood/mod.rs

```diff
@@ -26,6 +26,35 @@ pub struct BoweHopwoodPedersenCRH<G: Group, W: PedersenWindow, const N: usize> {
     window: PhantomData<W>,
 }
 
+impl<G: Group> BoweHopwoodPedersenParameters<G> {
+    pub fn from_generators(generators: Vec<Vec<G>>) -> Self {
```

Comment:
Let's call the check_preconditions() function here

Reply:
It cannot be called from here as it is a method of `BoweHopwoodPerdersenCRH` rather than `BoweHopwoodPedersenParameters`; I will move this function as a setup method of `BoweHopwoodPedersenCRH` to ensure that `check_preconditions` is called when instantiating the parameters